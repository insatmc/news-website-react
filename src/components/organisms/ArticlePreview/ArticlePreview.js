import React, { useContext } from "react"
import moment from "moment"
import PropTypes from "prop-types"
import { useTranslation } from "react-i18next"

import ArticleImage from "../../molecules/ArticleImage/ArticleImage"
import Text from "../../atoms/Text/Text"
import Link from "../../atoms/Link/Link"

import { store } from "../../../stores/articlesStore"

import "../../templates/ArticlePreviewTemplate/ArticlePreviewTemplate.scss"

export default function ArticlePreview({ article }) {
  const { t } = useTranslation()

  const { dispatch } = useContext(store)

  const onRemove = () => {
    dispatch({
      type: "REMOVE_ARTICLE",
      article: article.url,
    })
  }

  return (
    <div className="articlePreview">
      <Link to={article.url}>
        <ArticleImage
          onRemove={onRemove}
          title={article.title}
          image={article.urlToImage}
        />
      </Link>
      <div className="articleInfos">
        <Text size="m" kind="italic" color="#666">
          {moment(article.publishedAt).fromNow()} - {article.source.name}
        </Text>
      </div>
      <div className="articleDesc">
        <Text size="m">{article.description}</Text>
      </div>
      <div className="articleInfos">
        <Text>
          <Link to={article.url}>{t("Read more")}</Link>
        </Text>
      </div>
    </div>
  )
}

export const ArticleShape = PropTypes.shape({
  title: PropTypes.string,
  url: PropTypes.string,
  publishedAt: PropTypes.string,
  description: PropTypes.string,
  urlToImage: PropTypes.string,
  source: PropTypes.shape({
    name: PropTypes.string,
  }),
})

ArticlePreview.propTypes = {
  article: ArticleShape.isRequired,
}
