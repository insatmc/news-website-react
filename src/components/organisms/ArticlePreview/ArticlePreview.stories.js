import React from "react"
import ArticlePreview from "./ArticlePreview"

export const Default = () => (
  <ArticlePreview
    article={{
      title: "Title of article",
      urlToImage:
        "https://lh3.googleusercontent.com/w5zAdJ2zDj8ActvZEVESIqL0pevo1s4B7uzLhjn8bc9LRqPo5QM78OzLyCqjcDNYa6w",
      description:
        "The various threat intelligence stories in this iteration of the Weekly Threat Briefing discuss the following topics: APT, Backdoor, GoldenSpy, Phishing, Ransomware, and Vulnerabilities. The IOCs related to these stories are attached to the Weekly Threat Brieâ€¦",
      publishedAt: "2020-08-25T14:30:00Z",
    }}
  />
)

export default {
  title: "Components/Molecules/ArticlePreview",
  component: ArticlePreview,
}
