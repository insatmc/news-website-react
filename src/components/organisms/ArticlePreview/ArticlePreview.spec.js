import React from "react"
import { render } from "@testing-library/react"
import ArticlePreview from "./ArticlePreview"

describe("Article Preview", () => {
  it("should match snapshot", () => {
    const { wrapper } = render(
      <ArticlePreview
        article={{
          title: "Title of article",
          url: "path/to/article",
          urlToImage:
            "https://lh3.googleusercontent.com/w5zAdJ2zDj8ActvZEVESIqL0pevo1s4B7uzLhjn8bc9LRqPo5QM78OzLyCqjcDNYa6w",
          description:
            "The various threat intelligence stories in this iteration of the Weekly Threat Briefing discuss the following topics: APT, Backdoor, GoldenSpy, Phishing, Ransomware, and Vulnerabilities. The IOCs related to these stories are attached to the Weekly Threat Brieâ€¦",
          publishedAt: "2020-08-25T14:30:00Z",
          source: {
            name: "Source of news",
          },
        }}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
