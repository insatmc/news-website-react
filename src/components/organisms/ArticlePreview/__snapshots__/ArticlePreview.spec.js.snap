// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Preview should match snapshot 1`] = `undefined`;
